src/ngx_http_lua_rewriteby.c

```diff
@@ -164,24 +164,62 @@ ngx_http_lua_rewrite_handler(ngx_http_request_t *r)
 
 
 ngx_int_t
-ngx_http_lua_rewrite_handler_inline(ngx_http_request_t *r)
+ngx_http_lua_rewrite_handler_sets(ngx_http_request_t *r)
 {
-    lua_State                   *L;
     ngx_int_t                    rc;
     ngx_http_lua_loc_conf_t     *llcf;
+    ngx_array_t                 *p_sets;    /*   rewrite sets, inline
+                                                script/script file path */
+    ngx_uint_t                   i;
+    ngx_http_lua_rewrites_t     *rewrite;
 
-    dd("rewrite by lua inline");
+    dd("rewrite by lua handler sets");
+
+    rc = NGX_DECLINED;
 
     llcf = ngx_http_get_module_loc_conf(r, ngx_http_lua_module);
+
+    p_sets = llcf->rewrites;
+
+    for (i = 0; i < p_sets->nelts; ++i)
```

Comment:
This does not look quite right. Since a rewrite phase handler can yield and resume. And your loop always starts running from the first lua rewrite phase handler, which is wrong (when the 2nd handler yields and resumes, for example).